public/pages/workflow_detail/workspace/resizable_workspace.tsx

```diff
@@ -21,48 +32,148 @@ const COMPONENT_DETAILS_PANEL_ID = 'component_details_panel_id';
     (id: string, options: { direction: 'left' | 'right' }) => {}
   );
-
   const onToggleChange = () => {
     collapseFn.current(COMPONENT_DETAILS_PANEL_ID, { direction: 'left' });
     setIsOpen(!isOpen);
   };
 
+  // Selected component state
+  const { reactFlowInstance } = useContext(rfContext);
+  const [selectedComponent, setSelectedComponent] = useState<
+    ReactFlowComponent
+  >();
+
+  /**
+   * Hook provided by reactflow to listen on when nodes are selected / de-selected.
+   * - populate panel content appropriately
+   * - open the panel if a node is selected and the panel is closed
+   * - it is assumed that only one node can be selected at once
+   */
+  useOnSelectionChange({
+    onChange: ({ nodes, edges }) => {
+      if (nodes && nodes.length > 0) {
+        setSelectedComponent(nodes[0]);
+        if (!isOpen) {
+          onToggleChange();
+        }
+      } else {
+        setSelectedComponent(undefined);
+      }
+    },
+  });
+
+  useEffect(() => {
+    reactFlowInstance?.setNodes((nodes: ReactFlowComponent[]) =>
+      nodes.map((node) => {
+        node.data = {
+          ...node.data,
+          selected: node.id === selectedComponent?.id ? true : false,
+        };
+        return node;
+      })
+    );
+  }, [selectedComponent]);
+
+  // Formik form state
+  const [formValues, setFormValues] = useState<WorkspaceFormValues>({});
+  const [formSchema, setFormSchema] = useState<WorkspaceSchema>(yup.object({}));
+
+  // Initialize the form state to an existing workflow, if applicable.
+  useEffect(() => {
+    if (props.workflow?.workspaceFlowState) {
+      const initFormValues = {} as WorkspaceFormValues;
+      const initSchemaObj = {} as WorkspaceSchemaObj;
+      props.workflow.workspaceFlowState.nodes.forEach((node) => {
+        initFormValues[node.id] = componentDataToFormik(node.data);
+        initSchemaObj[node.id] = getComponentSchema(node.data);
+      });
+      const initFormSchema = yup.object(initSchemaObj) as WorkspaceSchema;
+      setFormValues(initFormValues);
+      setFormSchema(initFormSchema);
+    }
+  }, [props.workflow]);
+
+  // Update the form values and validation schema when a node is added
+  // or removed from the workspace
+  function onNodesChange(nodes: ReactFlowComponent[]): void {
+    const updatedComponentIds = nodes.map((node) => node.id);
+    const existingComponentIds = Object.keys(formValues);
+    const updatedSchemaObj = {} as WorkspaceSchemaObj;
+
+    if (updatedComponentIds.length > existingComponentIds.length) {
+      // TODO: implement for when a node is added
+    } else if (updatedComponentIds.length < existingComponentIds.length) {
+      existingComponentIds.forEach((existingId) => {
+        if (updatedComponentIds.includes(existingId)) {
+          updatedSchemaObj[existingId] = formSchema.fields[
+            `${existingId}`
+          ] as yup.ObjectSchema<any, any, any>;
+        } else {
+          delete formValues[`${existingId}`];
+        }
```

Comment:
For readability I can try to create some more temp variables, like deep copying the form values. I didn't do that initially since it wasn't necessary, and prevents having to make an extra `setFormValues()` call. Lmk what you think.